raiden-dapp/src/components/AddressInput.vue

```diff
@@ -100,41 +117,186 @@ export default class AddressInput extends Mixins(BlockieMixin) {
     // as the input being invalid. Since the :rules prop does not support
     // async rules we have to workaround with a reactive prop
     const isAddressValid =
-      !this.busy && this.errorMessages.length === 0 && this.isAddressAvailable;
+      !this.busy &&
+      !this.typing &&
+      this.errorMessages.length === 0 &&
+      this.isAddressAvailable;
+
     if (isAddressValid) {
       this.input(this.address);
     }
 
     return [() => isAddressValid || ''];
   }
 
+  private checkAvailability(
+    value: ValidationResult
+  ): Observable<ValidationResult> {
+    return iif(
+      () => 'error' in value,
+      of(value),
+      of(value).pipe(
+        tap(() => (this.busy = true)),
+        switchMap(value => {
+          return iif(
```

Comment:
`iif`, as `defer`, is relevant if you wanna define an observable through an expression which should be evaluated only at subscription time. `switchMap`'s callback (like `mergeMap`) is called and subscribed immediately when value flows through *Map's pipeline, therefore, there's no need of an `iif` or `defer` inside a merge or switchMap, and you can plainly use an `if` or a conditional ternary operator to choose the observable that should be merged to the output.

Reply:
Sounds reasonable I will modify accordingly thank you for the input :)